persist/fs/commitlog/reader.go

```diff
@@ -165,16 +292,11 @@ func (r *reader) readChunk() ([]byte, error) {
 		return nil, err
 	}
 
-	// Extend buffer as necessary
-	if len(r.dataBuffer) < int(size) {
-		diff := int(size) - len(r.dataBuffer)
-		for i := 0; i < diff; i++ {
-			r.dataBuffer = append(r.dataBuffer, 0)
-		}
-	}
+	// TODO: Pool this
+	newBuf := make([]byte, size, size)
```

Comment:
Yeah we'll definitely have to pool or else the GC pressure will be intense.

Reply:
The old code reused a single buffer over and over again (because there was no parallelism). I changed it to reuse a single buffer per "worker". I think that should be more efficient than pooling